hack/jenkins/e2e-runner.sh

```diff
@@ -208,21 +224,28 @@ if [[ "${E2E_UP,,}" == "true" ]]; then
     fi
 fi
 
-### Run tests ###
-# Jenkins will look at the junit*.xml files for test failures, so don't exit
-# with a nonzero error code if it was only tests that failed.
-if [[ "${E2E_TEST,,}" == "true" ]]; then
-    # Check to make sure the cluster is up before running tests, and fail if it's not.
-    go run ./hack/e2e.go ${E2E_OPT:-} -v --isup
-    go run ./hack/e2e.go ${E2E_OPT:-} -v --test \
-      ${GINKGO_TEST_ARGS:+--test_args="${GINKGO_TEST_ARGS}"} \
-      && exitcode=0 || exitcode=$?
-    if [[ "${E2E_PUBLISH_GREEN_VERSION:-}" == "true" && ${exitcode} == 0 && -n ${build_version:-} ]]; then
-        echo "Publish build_version to ci/latest-green.txt: ${build_version}"
-        gsutil cp ./version gs://kubernetes-release/ci/latest-green.txt
+# Allow download & unpack of alternate version of tests, for cross-version & upgrade testing.
+if [[ -n "${JENKINS_PUBLISHED_CROSS_VERSION:-}" ]]; then
+    cd ..
+    mv kubernetes kubernetes_old
+    fetch_published_version_tars "${JENKINS_PUBLISHED_CROSS_VERSION}"
+    cd kubernetes
+    # Upgrade the cluster before running other tests
+    if [[ -n "${GINKGO_UPGRADE_TEST_ARGS:-}" ]]; then
```

Comment:
It's probably fine as-is.

We're actually pretty close to being able to `source e2e-runner.sh && fetch_published_version_tars && unpack_tars && (up) && e2e_test && (down)`.